reposerver/repository/helm.go

```diff
@@ -0,0 +1,112 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	argoexec "github.com/argoproj/pkg/exec"
+
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/reposerver/apiclient"
+	"github.com/argoproj/argo-cd/util"
+	"github.com/argoproj/argo-cd/util/helm"
+)
+
+func (s *Service) GetHelmCharts(ctx context.Context, q *apiclient.HelmChartsRequest) (*apiclient.HelmChartsResponse, error) {
+	index, err := helm.GetIndex(q.Repo.Repo, q.Repo.Username, q.Repo.Password)
+	if err != nil {
+		return nil, err
+	}
+	res := apiclient.HelmChartsResponse{}
+	for chartName, entries := range index.Entries {
+		chart := apiclient.HelmChart{
+			Name: chartName,
+		}
+		for _, entry := range entries {
+			chart.Versions = append(chart.Versions, entry.Version)
+		}
+		res.Items = append(res.Items, &chart)
+	}
+	return &res, nil
+}
+
+func (s *Service) helmChartRepoPath(repo string) (string, error) {
+	repoPath := tempRepoPath(repo)
+
+	s.repoLock.Lock(repoPath)
+	defer s.repoLock.Unlock(repoPath)
+
+	err := os.Mkdir(repoPath, 0700)
```

Comment:
I seem to recall someone concerned about permissions, is `0700` correct?

Reply:
I think that was this comment: https://github.com/argoproj/argo-cd/pull/1865#discussion_r319433791 . The suggestion was to change `0777` to `0700`